docs/01 - Running & Configuring the Application/A - Running & Configuring the Application.md

```diff
@@ -0,0 +1,154 @@
+
+##### Yaml Search
+
+The Platform configuration loader tries to fetch the configuration from up to three different sources before failing, in the following order:
+
+1. The JVM Option `-Dstellar.anchor.config`, for instance:
+
+    ```shell
+    ./gradlew service-runner:bootRun --args=--sep-server -PjvmArgs="-Dstellar.anchor.config=[path-to-yaml]" 
+    ```
+
+2. The file `.anchor/anchor-config.yaml` in the user's home directory. If the path of the `yaml` is not specified by the JVM options, the server will search for the `./anchor/anchor-config.yaml` file in the user's home directory.
+
+3. The system Environment Variable `STELLAR_ANCHOR_CONFIG`. If neither the JVM option nor the file `.anchor/anchor-config.yaml` is found, the server will search for the `STELLAR_ANCHOR_CONFIG`, whose default value is `"classpath:/anchor-config-defaults.yaml"`:
+
+    ```shell
+    STELLAR_ANCHOR_CONFIG=classpath:/anchor-config-defaults.yaml
+    ./gradlew service-runner:bootRun --args=--sep-server
+    ```
+
+If all of the above fail, the server will fail with an error.
+
+### Environment variables
+
+Secrets are passed to the Anchor Platform (and Anchor Reference Server) via environment variables, which can be set either through command line or using a `.env` file. A list of supported environment variables and their descriptions can be found at [`example.env`].
+
+### Event Messaging
+
+A message queue is required for the Anchor Platform to publish messages to and for the Anchor to consume messages from.
+The default queueType used (in the anchor-config.yaml file) is "kafka", and SQS is also supported.
+
+The default Kafka configuration should work out of the box if you have a running Kafka available at the correct port. The easiest way for that is by using the docker-compose.yaml file located at `docs/resources/docker-examples/kafka/docker-compose.yaml`:
+
+```shell
+cd docs/resources/docker-examples/kafka
+docker compose up
+```
+
+### JVM-Argument based run-configuration
+
+Since Java processes take `-D` arguments as JVM system properties, the path/locator of the `yaml` file can be passed to the process through JVM system properties:
+
+```shell
+java -Dstellar.anchor.config=file:/path/to/file.yaml -jar anchor-platform.jar --anchor-reference-server
+```
+
+## Docker
+
+Docker Build:
+
+```shell
+docker build -t stellar-anchor-platform:latest .
+```
+
+Docker Run:
+
+Note: secrets (credentials, tokens, etc...) are passed to the application via environment variables. Use `-e` to pass in
+each required environment variables ([Environment Variables](/platform/src/main/resources/example.env))
+
+```shell
+docker run -v {/local/path/to/config/file/}:/config -p 8081:8081 stellar-anchor-platform:latest --anchor-reference-server \
+-e JWT_SECRET='secret' \
+-e SEP10_SIGNING_SEED='SAX3...C3AW5X' \
+-e 'CIRCLE_API_KEY=QVBJX0...NjMyZTQ5NWJhNDdlZg==' \
+-e PAYMENT_GATEWAY_STELLAR_SECRET_KEY='secret' \
+-e POSTGRES_USERNAME='postgres' \
+-e POSTGRES_PASSWORD='password'
```

Comment:
Are these env vars up to date?

Reply:
Yup, they are. There are other envs that could be used too, depending on the setup (like sqlite or sqs for queues).

Reply:
I've added a note about that in https://github.com/stellar/java-stellar-anchor-sdk/pull/305/commits/b97a6b68badd79d58e613d0cc7133b753db75199.